lib/Sema/CSSimplify.cpp

```diff
@@ -4497,7 +4504,280 @@ ConstraintSystem::simplifyApplicableFnConstraint(
     return simplified;
   }
 
-  return SolutionKind::Error;
+  // Handle applications of @dynamicCallable types.
+  return simplifyDynamicCallableApplicableFnConstraint(type1, origType2,
+                                                       subflags, locator);
+}
+
+/// Looks up and returns the @dynamicCallable required methods (if they exist)
+/// implemented by a type.
+static llvm::DenseSet<FuncDecl *>
+lookupDynamicCallableMethods(Type type, ConstraintSystem &CS,
+                             const ConstraintLocatorBuilder &locator,
+                             Identifier argumentName, bool hasKeywordArgs) {
+  auto &ctx = CS.getASTContext();
+  auto decl = type->getAnyNominal();
+  auto methodName = DeclName(ctx, ctx.Id_dynamicallyCall, { argumentName });
+  auto matches = CS.performMemberLookup(ConstraintKind::ValueMember,
+                                        methodName, type,
+                                        FunctionRefKind::SingleApply,
+                                        CS.getConstraintLocator(locator),
+                                        /*includeInaccessibleMembers*/ false);
+  // Filter valid candidates.
+  auto candidates = matches.ViableCandidates;
+  auto filter = [&](OverloadChoice choice) {
+    auto cand = cast<FuncDecl>(choice.getDecl());
+    return !isValidDynamicCallableMethod(cand, decl, CS.TC, hasKeywordArgs);
+  };
+  candidates.erase(std::remove_if(candidates.begin(), candidates.end(), filter),
```

Comment:
Here too :)

Reply:
Very minor, but:

This line was actually at 80 columns long, actually (i.e. the last character was on column 80).
I wonder if the 80 column convention is:
* The last character of a line can be on column 80 (or before). I've been following this.
* The last character of a line can be on column 79 (or before). I suppose this makes the line "fit within 80 columns".

Anyhow, I changed this line and other lines that end on column 80 so that they end on column 79.